Realm.Shared/RealmObjectOps.cs

```diff
@@ -0,0 +1,393 @@
+﻿using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace Realms
+{
+    [Preserve(AllMembers = true)]
+    internal static class RealmObjectOps
```

Comment:
Ops as in "operations"? Why are all these duplicated methods necessary, is there really no way to avoid that?